Sources/FontMetrics.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  FontMetrics.swift
+//  Relativity
+//
+//  Created by Dan Federman on 12/26/16.
+//  Copyright © 2016 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import CoreGraphics
+import UIKit
+
+
+public struct FontMetrics {
+    
+    // MARK: Initialization
+    
+    public init(for font: UIFont) {
+        self.font = font
+    }
+    
+    // MARK: Public Properties
+    
+    public let font: UIFont
+    
+    public var topInset: CGFloat {
+        return PixelRounder().floorToPixel(font.ascender - font.capHeight)
+    }
+    
+    public var bottomInset: CGFloat {
+        return PixelRounder().ceilToPixel(abs(font.descender))
```

Comment:
It's problematic that this creates and uses a default PixelRounder internally, and a little odd that it's a struct wrapping a UIFont instead of an extension on UIFont. Also this should be `-font.descender` instead of `abs` since a font with a positive descender shouldn't be treated like one with a negative descender of the same magnitude.